test/spec/common/client/addAlgoliaAgent.js

```diff
@@ -40,3 +40,44 @@ test('client.addAlgoliaAgent(algoliaAgent)', function(t) {
     fauxJax.restore();
   });
 });
+
+test('custom search time agent with x-algolia-agent', function(t) {
+  t.plan(1);
+
+  var fauxJax = require('faux-jax');
+  var parse = require('url-parse');
+  fauxJax.install({gzip: true});
+
+  var createFixture = require('../../../utils/create-fixture');
+  var fixture = createFixture();
+
+  var client = fixture.client;
+  var index = fixture.index;
+
+  client.addAlgoliaAgent('And some other incredible agent');
+
+  // Ensure we de-duplicate by re-adding the same agent a second time.
+  client.addAlgoliaAgent('And some other incredible agent');
+
+  index.search('algolia agent', {
+    'x-algolia-agent': 'the other agent'
+  });
+
+  var expectedAgent = fixture.algoliasearch.ua + ';And some other incredible agent';
```

Comment:
I would have expected the agent to be:

fixture.algoliasearch.ua + ';the other agent'. I cannot see where do we test for the presence of "the other agent"